src/gluonts/model/deepstate/_estimator.py

```diff
@@ -166,15 +168,13 @@ def __init__(
             num_eval_samples > 0
         ), "The value of `num_eval_samples` should be > 0"
         assert dropout_rate >= 0, "The value of `dropout_rate` should be >= 0"
-        assert (cardinality is not None and use_feat_static_cat) or (
-            cardinality is None and not use_feat_static_cat
-        ), "You should set `cardinality` if and only if `use_feat_static_cat=True`"
-        assert cardinality is None or [
-            c > 0 for c in cardinality
-        ], "Elements of `cardinality` should be > 0"
-        assert embedding_dimension is None or [
-            e > 0 for e in embedding_dimension
-        ], "Elements of `embedding_dimension` should be > 0"
+        assert not use_feat_static_cat or np.prod(cardinality) > 1, (
```

Comment:
I like `all(c > 1 for c in cardinality)` better since it's more explicit. With this change the `import numpy as np` also goes

Reply:
But that is not equivalent to the condition we want: we should still allow one of  the cardinalities to be equal to 1 if there are more than one categorical feature (e.g., see smoke test). This, for example, is useful if one wants to experiment on some slice of data by fixing one of the categorical features without changing the dataset.

I should have written this with `any` as you mentioned offline.